etl-postgres/src/replication/slots.rs

```diff
@@ -1,41 +1,138 @@
 use sqlx::PgPool;
+use thiserror::Error;
+use tokio_postgres::types::Oid;
 
-use crate::replication::worker::WorkerType;
 use crate::types::TableId;
 
 /// Maximum length for a Postgres replication slot name in bytes.
 const MAX_SLOT_NAME_LENGTH: usize = 63;
 
 /// Prefixes for different types of replication slots
-const APPLY_WORKER_PREFIX: &str = "supabase_etl_apply";
-const TABLE_SYNC_PREFIX: &str = "supabase_etl_table_sync";
+pub const APPLY_WORKER_PREFIX: &str = "supabase_etl_apply";
+pub const TABLE_SYNC_WORKER_PREFIX: &str = "supabase_etl_table_sync";
 
 /// Error type for slot operations.
-#[derive(Debug, thiserror::Error)]
-pub enum SlotError {
+#[derive(Debug, Error)]
+pub enum EtlReplicationSlotError {
     #[error("Invalid slot name length: {0}")]
     InvalidSlotNameLength(String),
+
+    #[error("Invalid slot name: {0}")]
+    InvalidSlotName(String),
+}
+
+/// Parsed representation of a replication slot name.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum EtlReplicationSlot {
+    /// Apply worker slot for a pipeline.
+    Apply { pipeline_id: u64 },
+    /// Table sync worker slot for a pipeline and table.
+    TableSync { pipeline_id: u64, table_id: TableId },
 }
 
-/// Generates a replication slot name for the given pipeline ID and worker type.
-pub fn get_slot_name(pipeline_id: u64, worker_type: WorkerType) -> Result<String, SlotError> {
-    let slot_name = match worker_type {
-        WorkerType::Apply => {
-            format!("{APPLY_WORKER_PREFIX}_{pipeline_id}")
+impl EtlReplicationSlot {
+    /// Creates a new [`EtlReplicationSlot`] for the apply worker.
+    pub fn for_apply_worker(pipeline_id: u64) -> Self {
+        Self::Apply { pipeline_id }
+    }
+
+    /// Creates a new [`EtlReplicationSlot`] for the table sync worker.
+    pub fn for_table_sync_worker(pipeline_id: u64, table_id: TableId) -> Self {
+        Self::TableSync {
+            pipeline_id,
+            table_id,
         }
-        WorkerType::TableSync { table_id } => {
-            format!(
-                "{TABLE_SYNC_PREFIX}_{pipeline_id}_{}",
-                table_id.into_inner()
-            )
+    }
+
+    /// Generates a replication slot name for the given pipeline ID and worker type.
+    pub fn try_to_string(&self) -> Result<String, EtlReplicationSlotError> {
```

Comment:
Instead of `try_to_string` and `try_from_string` why not implement `TryFrom` for appropriate types?